### PR TITLE
PCVL 677 refactor pauli circuits

### PR DESCRIPTION
**Pull request by raksharuia**

### CODE REVIEW

File: perceval/components/_pauli.py [hunk 1]
```diff
@@ -31,131 +31,108 @@
...
 from .unitary_components import BS, PS, PERM
-from .processor import Processor
 from perceval.utils import Matrix
-from typing import List
 
 
 class PauliType(Enum):
+    """
+    Enumeration of different Pauli Types. Usage -
+    1. Choose a Pauli gate (/operator)
```
Comment (ericbrts) [inline perceval/components/_pauli.py:40]: I can't see how your "Usage + 1. and 2. items" correspond to the enum documentation. The enum is just a set of named values, the doc (if needed), should explain what the I, X, Y, Z name stand for.

I'm not asking to document this enum, just to move these few lines of doc to a more appropriate location.

File: perceval/components/_pauli.py [hunk 2]
```diff
@@ -31,131 +31,108 @@
...
 
-    return p
+    if pauli_type == PauliType.I:
+        return Circuit(2, name="I Measurer")
+    elif pauli_type == PauliType.X:
+        return Circuit(2, name="Pauli X Measurer") // BS.H()
+    elif pauli_type == PauliType.Y:
+        return Circuit(2, name="Pauli Y Measurer") // BS.Rx(theta=np.pi/2, phi_bl=np.pi, phi_br=-np.pi/2)
+    elif pauli_type == PauliType.Z:
+        return Circuit(2, name="Pauli Z Measurer ")
```
Comment (ericbrts) [inline perceval/components/_pauli.py:136]: Remove trailing space

File: tests/test_pauli.py [hunk 3]
```diff
@@ -41,9 +41,10 @@ def test_pauli_type():
     assert is_ascending == True
 
 
-@pytest.mark.parametrize("pauli_gate", [PauliType.I, PauliType.X, PauliType.Y, PauliType.Z])
-def test_pauli_state_prep_circuits(pauli_gate):
-    c = Circuit(2) // get_preparation_circuit(pauli_gate)
+#@pytest.mark.parametrize("pauli_gate", [PauliType.I, PauliType.X, PauliType.Y, PauliType.Z])
```
Comment (ericbrts) [inline tests/test_pauli.py:44]: Remove commented line

**Review by ericbrts — APPROVED** [review]

File: tests/test_pauli.py (hunk 3 above)
Comment (raksharuia) [reply tests/test_pauli.py:44]: oops.

**Pull request merged**